src/test/java/io/appform/dropwizard/sharding/dao/LookupDaoTest.java

```diff
@@ -180,6 +184,25 @@ public void testScatterGather() throws Exception {
                         .getText());
     }
 
+    @Test
+    public void testScatterGatherWithQuerySpec() throws Exception {
+        List<TestEntity> results = lookupDao
+                .scatterGather((queryRoot, query, criteriaBuilder)
+                        -> query.where(criteriaBuilder.equal(queryRoot.get("externalId"), "testId")));
+        assertTrue(results.isEmpty());
+        TestEntity testEntity = TestEntity.builder()
+                .externalId("testId")
+                .text("Some Text")
+                .build();
+        lookupDao.save(testEntity);
+        results = lookupDao.scatterGather(DetachedCriteria.forClass(TestEntity.class)
+                .add(Restrictions.eq("externalId", "testId")));
+        assertFalse(results.isEmpty());
+        assertEquals("Some Text",
+                results.get(0)
+                        .getText());
+    }
+
```

Comment:
Tests needed for all newly introduced functions

Reply:
Done